### ISSUE TITLE
Atomic data files - directories

### ISSUE THREAD

**Issue by kslong** [sirocco-rt/sirocco#623]
We need to decide how to handle the fact that we will need to share among ourselves atomic data that is not necessarily part of the regular distribution, e.g for example macro atom files that we are testing.  

We need to think this through carefully, so that on the one hand we do not make our lives harder as we are working, and on the other hand make it clear to our external users that these are "in test" as it were.

**Comment by kslong** [comment]
@Higginbottom has attempted to re-organize the atomic data directories as discussed in #614.  However the current data structure is currently still quite complicated and does not (from ksl's perspective) look simple.  I believe we somehow need to be able to reorganize the files so that is simple to use old versions of the data, or slightly modified versions of them anyway, but make the main data directory have a cleaner look.  

What would be nice if we could layout the data directory more like the examples directory, with subdirectories for current, old, and development atomic datasets.  

But the issues surrounding this are fairly complex and it is not obvious exactly what we want to do.  Part of the "problem" is that because of the fact that we have a "masterfile" approach, one of the issues is what happens if one of the underlying files is changed.  Also note that from the point of total data, it is the models that dominate.  The atomic data itself is relatively small, so whatever solution is adopted needs to take this into account

So I think we need to discuss the best way to proceed.

**Comment by kslong** [comment]
@Higginbottom  @jhmatthews   I have pushed a new version of the atomic data files to a branch called dev.  I would like you to look at this, and see whether it is something you think we should pursue.  At a top level, 

- all of the models are in subdirectory called models
- all of the currently standard atomic data files are in a subdirectory called base
- the CIII etc macro atom model standard files are in dev, since we are working to verify them
- the old atomic data is in old

I have not figured out everything would work in practice, but I am thinking that Setup_Py_Dir would take an optional argument to link the local data directory to either base (no argument needed), dev, or old.  This would clean up the data directories for standard users.

This is not completely OK for dev or possibly for old directories, because of the way we use master files to collect other files of atomic data together.  In principle, and perhaps this is good enough for us, we can use symbolic links to stitch things together, but that may not be the best approach.  (git does allow you to add symbolic links if you are careful.)

Regardiing the models, I would probably modify SetUp_Py_Dir to link this as well, and this would affect what needed for various links.  

An alternative approach which also bears thinking about would be to make SetUp_Py_Dir more sophisticated so that it constructs a set of symbolic links located locally to the right data files.  (A point related to all of this is that Setup_Py_Dir or its replacement should be part of the data repository, rather than the Python repository itself). 

At any rate, I would appreciate your looking at this structure and seeing if this approach is worth pursuing.  If not, I can just blow this branch away.  

P. S. if we were starting over, I would still advocate for putting the atomic data in with Python (even if we decided to keep the models, which are the bulk of the data separate).  Five years from now, if we are still working on Python, we will wish we had done this.

**Comment by saultyevil** [comment]
Just a thought, but instead of making Setup_Py_Dir more complicated/sophisticated, why don't we require data to be in the Python root directory? I think this is a similar approach to what Cloudy does - though I haven't looked at the code extensively.

When a user then specifies that they want to use, say, the standard80 atomic data, we could use `get_env("PYTHON")` to construct the correct directory path. This could done in `get_line_transfer_mode` where we call `get_atomic_data`, rather than in get_atomicdata itself to keep it Python agnostic. I think this way, we could probably keep our masterfile scheme without adding any extra complications and confusion for non-developer users.

**Comment by jhmatthews** [comment]
Ed's suggestion makes sense to me given that the $PYTHON variable is used by Setup_Py_Dir anyway. Although I don't have any objection to organise masterfiles into subfolders like Knox suggests - might make things cleaner.

**Comment by kslong** [comment]
I have begun to implement a version of what I suggested.  I have a version of the code that has a data directory (which I can easily incorporate into the Python repository) and a models directory (which is large, and which we agreed would be separate).  I have a couple of questions on how to proceed though.

*  What is the easiest way to make sure we can go back to using old datasets.  One possibility is to leave the data directory as it is, and determine a new name for the directory in which we want to put data going forward.  Note that if I do that, I might want to revert our old data directory to a time before we started to reorganize it.  The advantage of the approach is that all of our old runs will work as they did before.  An alternative is to switch this around, and to put the old files into a directory data_old for example, and then to fix up the .pf files ... but that seems more awkward to me
* We surely want our regression tests to still work, and probably all example .pf files.  What do we about cv_kur for example, which uses kurucz models?  One possibility here, is simply to say that we expect people to download the models repository, regardles.  It's just the atomic data that we want to put into the Python repository.   The other possibility is to put a limited subset of the models into our new repository, but this seems cumbersome ultimately.
* How do we separate atomic data files that we are working on but are not for general consumption from those that we think are validated?  The problem here for me is that our new datasets are often just a little new, and most of the data is actually in the same place as it always was.  Right now for example, the various "masterfiles" are all at the top level, and all of the atomic data is either in atomic or atomic_macro.  Is that OK?

Please send comments and thoughts on this.

@saultyevil  while, it may have some appeal, it's not obvious to me that  requiring that the data be sitting under $PYTHON is a good idea.  That would mean if someone were working on say TDEs with odd abundances that these would need to be put there.

**Comment by kslong** [comment]
@jhmatthews I am ready to merge the new version of the database files into Python.  But I need a new repository in AGNWinds and I don't know how to do this (probably because I don't know the appropriate way to log directly into AGNWinds.  Could you (outside of github obviously)  let me know what I need to do to login or alternatively simple create an empty repository with a name that provides an appropriate place to put large grids of stellar  and other spectra, e.g SpecGrid.  The name needs to be descriptive but short.

**Comment by kslong** [comment]
Changes have been merged into dev.
